library/network/test/network_service_test.rb

```diff
@@ -39,5 +46,52 @@
 
       subject.RunSystemCtl("wicked", "disable | evil")
     end
+
+    it "raises an exception when no service name is provided" do
+      expect { Yast::NetworkService.RunSystemCtl(nil, "enable") }.to raise_error
+    end
+  end
+
+  describe "#EnableDisableNow" do
+    subject { Yast::NetworkService }
+
+    context "When changing running service" do
+      before(:each) do
+        allow(subject).to receive(:current_name).and_return(:netconfig)
+
+        # using anything instead of exact service name because of some magic in identifying the service in the system
+        expect(subject).to receive(:RunSystemCtl).with(anything, /stop|kill/)
+        expect(subject).to receive(:RunSystemCtl).with("network", "disable")
+      end
+
+      it "disables old service and enables new one" do
+        allow(subject).to receive(:cached_name).and_return(:wicked)
+
+        expect(subject).to receive(:RunSystemCtl).with("wicked", "enable", any_args)
+
+        Yast::NetworkService.EnableDisableNow
+      end
+
+      it "only disables old service when no network service was requested" do
```

Comment:
and this could be a context where you declare the cached_name as nil

Reply:
i believe that there is many ways how to write tests ... and each of us prefer one of them ;-)

Reply:
Sure, but there are also conventions, style guidelines and good practices that we should follow as good citizens :stuck_out_tongue_winking_eye: 

http://www.betterspecs.org/

Reply:
I'm not nice guy today ;-) But seriously ... I've even problem with some rubocop checks, so convention is good ... but I break them time to time on purpose.